src/middleware/user.js

```diff
@@ -42,12 +42,12 @@ module.exports = function (middleware) {
 			return true;
 		}
 
-		if (req.loggedIn) {
+		if (res.locals.isAPI && (req.loggedIn || !req.headers.hasOwnProperty('authorization'))) {
```

Comment:
If the intended purpose of the change is to enforce CSRF token checking for guests, then why keep `req.loggedIn` at all? Just checking that `req.headers` doesn't contain `authorization` should cover all cases of requests that are cookie-based.

(Unless I am missing something...)

Reply:
It was to keep functionality exactly the same for loggedIn users. Previous code always applied csrf for loggedIn users even if req.headers had authorization. If it's fine to change that we can remove req.loggedIn from that if.